src/View.php

```diff
@@ -185,6 +185,15 @@ public function setModel(Model $model)
      */
     public function setSource(array $data, $fields = null)
     {
+        // ID with zero value is not supported, fix the data
+        if (isset($data[0])) {
```

Comment:
@georgehristov because of https://github.com/atk4/data/pull/806 when validation is triggered.

I am not sure with this. We can switch to mandatory or allow zero IDs for non SQL persistences only.

Do you know, if zero as ID is problem with some DB engine?

Reply:
I was able to reproduce http://sqlfiddle.com/#!9/d28469/1 - at least with MySQL zero is understood as empty ID and AI is applied.

Reply:
Sqlite hovewer supports zero - https://dbfiddle.uk/?rdbms=sqlite_3.27&fiddle=f2d14bf69f3a8d2ae85b3ff07703c9fa

Reply:
with last fix I think we can keep "required" (and thus reject zero IDs)